manifests/init.pp

```diff
@@ -1,21 +1,35 @@
 class wireguard (
-  Array[Hash] $tunnels = [],
-  Array[Hash] $simple_tunnels = [],
+  Hash[String, Struct[
+    {
+      ensure      => Optional[String],
+      private_key => String,
+      listen_port => Integer,
+      virtual_ip  => String,
+      peers       => Hash,
+    }
+  ]] $tunnels = {},
+  Hash[String, Struct[
+    {
+      ensure           => Optional[String],
+      private_key      => String,
+      listen_port      => Integer,
+      virtual_ip       => String,
+      peer_public_key  => String,
+      peer_allowed_ips => Optional[String],
+      peer_endpoint    => Optional[String],
+    }
+  ]] $simple_tunnels = [],
```

Comment:
In case you have an idea of a better name for `$simple_tunnels`, `wireguard::simple_tunnel`, etc. feel free to change that too. Idea is to this as a short-hand for point-to-point links (with only one peer)